### ISSUE TITLE
Incorrect report for `no-script-url `

### ISSUE THREAD

**Issue by MoOx** [eslint/eslint#2530]
I am making a script which securing some urls and remove `javascript:` (note that you should also handle the `vbscript:`, just in case), but when I do my test using `indexOf("javascript:")`, I am getting an error. I know this won't be easy to fix, but just wanted to let you know.

```js
var foo = bar.indexOf("javascript:");
```

- 1:22 - Script URL is a form of eval.

**Comment by nzakas** [comment]
This rule is very naive, and unfortunately that's just the nature of what it's trying to do. Is there a reason you don't turn it off?

**Comment by MoOx** [comment]
I disabled it indeed. Just want to let you know this rule can be weird :) But I clearly understand why. Feel free to close.
